app/services/topology_service_api.rb

```diff
@@ -9,4 +10,26 @@ def initialize(options)
     end
     @api_instance = TopologicalInventoryApiClient::DefaultApi.new
   end
+
+  def to_service_params
+    hashy = instance_variables.each_with_object({}) do |var, hash|
+      next if [:@params, :@api_instance].include?(var)
+      hash[var.to_s.delete("@")] = instance_variable_get(var)
+    end
+    hashy.compact
+  end
+
+  private
+
+  def apply_instance_vars(obj)
+    uniq_ivars(obj).each do |ivar|
+      value = obj.instance_variable_get(ivar)
+      instance_variable_set(ivar, value)
+    end
+    self
+  end
+
+  def uniq_ivars(object)
+    self.instance_variables & object.instance_variables
+  end
```

Comment:
Is there a reason that all these new methods are in this class instead of the `service_offering` class? Right now, it's only being used in the `service_offering` class.

I especially think that the `to_service_params` method should be moved since that is language that (to me) seems specific to the `service_offering` class. Plus, you're testing the `to_service_params` method in the `service_offering_spec` file, so that could be confusing if we leave it here. I think you have a case for leaving `apply_instance_vars` and `uniq_ivars` here, since they do look pretty generic. However, they aren't being used by anything the `TopologyServiceApi` is doing so it seems odd right now.

Reply:
The plan ( at least from what I'm seeing coming down the pipeline ) is to create a pattern where for every model we can control what normalized parameters are passed in from the services.  I added these methods into the base class because I expect more child classes to make use of them using the `instance_variable` pattern.  I did debate on separating out the pattern from this PR, and would be willing to so if that would clear up some confusion.

Reply:
Would like to push out the naming change to a follow up PR.  You have https://github.com/ManageIQ/service_portal-api/pull/31 and once we get consensus on naming services we can do a one off change.

Reply:
Ok, so, I guess I'm confused about the `to_service_params` method then. The way I read it is that it is a method that is generating parameters with regards to service offerings. I guess that's incorrect? Should it maybe be something more like `to_parameters_for_services`?

The problem I think lies in the fact that we have service objects (as in, classes that are in the `app/services` directory meant to facilitate logic for controllers), and we have actual objects that have the name `service` in them. Does that make sense? Just trying to clarify where exactly this method should go.